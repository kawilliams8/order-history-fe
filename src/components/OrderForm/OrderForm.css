.OrderForm {
  width: 350px;
}

.nameInput,
.descriptionInput,
.priceInput,
.submitButton {
  height: 25px;
  margin: 10px;
  width: 300px;
}

.submitButton {
  background-color: #5fdb6f;
  color: white;
}