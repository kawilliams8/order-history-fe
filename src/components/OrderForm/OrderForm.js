import React, {Component} from 'react';
import '../OrderForm/OrderForm.css';
import notFound from '../../images/notfound.png'

class OrderForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      img: "",
      name: "",
      description: "",
      price: ""
    }

  }

  handleTyping = (e) => {
    this.setState({
      [e.target.name] : e.target.value
    })
  }

  handleFormSubmit = (e) => {
    e.preventDefault()
    const newOrder = {
      ...this.state,
      id: Date.now(),
      img: {notFound}
    }
    const { addOrder } = this.props;
    addOrder(newOrder);
    this.setState({
      img: "",
      name: "",
      description: "",
      price: ""
    })
  }

  render() {
    return (
      <form className="OrderForm">
        <input className="nameInput" type="text" name="name" placeholder="Item Name" value={this.state.name} onChange={this.handleTyping}></input>
        <input className="descriptionInput" type="text" name="description" placeholder="Item Description" value={this.state.description} onChange={this.handleTyping}></input>
        <input className="priceInput" type="number" name="price" placeholder="Item Price" value={this.state.price} onChange={this.handleTyping}></input>
        <button className="submitButton" onClick={this.handleFormSubmit}>Add Purchase</button>
      </form>
    )
  }
}

export default OrderForm;