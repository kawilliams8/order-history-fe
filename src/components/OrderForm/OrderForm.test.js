import React from 'react';
import { shallow } from 'enzyme';
import OrderForm from '../OrderForm/OrderForm';

describe("Order Form", () => {
  let wrapper;
  beforeEach(() => {
    const addOrderMock = jest.fn()
    wrapper = shallow(<OrderForm addOrder={addOrderMock} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should update Form\'s state when typing', () => {
    const mockEvent = {target: {name: "name", value: "Item Name"}};
    const expected = "Item Name";
    wrapper.instance().handleTyping(mockEvent);
    expect(wrapper.state('name')).toEqual(expected);
  });

  it('should call the addOrder method when you click', () => {
    wrapper.instance().addOrder = jest.fn();
    const mockEvent = {preventDefault: jest.fn()};
    wrapper.instance().forceUpdate();
    wrapper.find('button').simulate('click', mockEvent);
    expect(wrapper.instance().addOrder).toHaveBeenCalled();
  })
})