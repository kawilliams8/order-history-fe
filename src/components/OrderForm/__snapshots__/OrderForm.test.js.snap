// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Form should match the snapshot 1`] = `
<form
  className="OrderForm"
>
  <input
    className="imageInput"
    name="img"
    onChange={[Function]}
    placeholder="Item Image URL"
    type="text"
    value=""
  />
  <input
    className="nameInput"
    name="name"
    onChange={[Function]}
    placeholder="Item Name"
    type="text"
    value=""
  />
  <input
    className="descriptionInput"
    name="description"
    onChange={[Function]}
    placeholder="Item Description"
    type="text"
    value=""
  />
  <input
    className="priceInput"
    name="price"
    onChange={[Function]}
    placeholder="Item Price"
    type="number"
    value=""
  />
  <button
    className="submitButton"
    onClick={[Function]}
  >
    Add Purchase
  </button>
</form>
`;
